static/js/components/search/NestedAggregatingMenuFilter.js

```diff
@@ -0,0 +1,292 @@
+    let filterElement = R.pathOr({}, ['nested', 'filter'], appliedNestedFilters);
+    if (R.path(['bool', 'must'], filterElement)) {
+      return R.path(['bool', 'must'], filterElement);
+    } else if (filterElement.term) {
+      return [filterElement];
+    } else {
+      return [];
+    }
+  }
+
+  /**
+   * Creates an 'AND' filter for all filters that are applied on this element's nested path, minus the filter for
+   * this specific element (if it exists). If no other filters are applied on this element's nested path, undefined is
+   * returned.
+   *
+   * Example return value: {
+   *   'bool': {
+   *     'must': [
+   *       {'term': {'program.enrollments.course_title': 'Some Course Title'}
+   *     ]
+   *   }
+   * }
+   */
+  createFilterForOtherElementsOnPath(query) {
+    let allTermFilters = this.getAllTermFiltersOnPath(query);
+    let otherTermFilters = [];
+    if (allTermFilters.length > 0) {
+      // Filter out term filters for this element
+      otherTermFilters = R.filter(
+        R.compose(
+          R.not,
+          R.equals(this.key),
+          R.head,
+          R.keys,
+          R.prop('term')
+        )
+      )(allTermFilters);
+    }
+    return otherTermFilters.length > 0 ? BoolMust(otherTermFilters) : undefined;
+  }
+
+  /**
+   * Gets the appropriate terms bucket for this element's agg query.
+   */
+  getTermsBucket(query) {
+    let otherAppliedFiltersOnPath = this.createFilterForOtherElementsOnPath(query);
+    let termsKey = otherAppliedFiltersOnPath ? INNER_TERMS_AGG_KEY : this.key;
+    let termsBucket = ReverseNestedTermsBucket(termsKey, this.key, _.omitBy({
+      size:this.size,
+      order:this.getOrder(),
+      include: this.options.include,
+      exclude: this.options.exclude,
+      min_doc_count:this.options.min_doc_count
+    }, _.isUndefined));
+
+    if (otherAppliedFiltersOnPath) {
+      return FilterBucket(
+        this.key,
+        otherAppliedFiltersOnPath,
+        termsBucket
+      );
+    } else {
+      return termsBucket;
+    }
+  }
+}
+
+export default class NestedAggregatingMenuFilter extends PatchedMenuFilter {
+  /**
+   * Overrides defineAccessor in MenuFilter
+   * Sets a custom Accessor for this Filter type. This is otherwise identical to the original implementation.
+   */
+  defineAccessor() {
+    return new NestedAggregatingFacetAccessor(
+      this.props.field, this.getAccessorOptions()
+    );
+  }
+
+  /**
+   * Overrides getItems in MenuFilter
+   * Before the aggregation results are rendered, set the doc_count of each item to be the
+   * "reverse nested" doc_count. This effectively means that we will show how many unique users
+   * match the query against a set of nested elements, as opposed to the total count of nested
+   * elements that match (which could be greater than the number of users).
+   */
+  getItems() {
+    let items = super.getItems();
+    items = _.map(items, (item) => {
```

Comment:
This will modify each item which is passed in. Can we use a `_.forEach` instead to make the mutation explicit?